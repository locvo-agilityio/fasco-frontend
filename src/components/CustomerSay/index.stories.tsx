import type { Meta, StoryObj } from '@storybook/react';

// Components
import CustomerSay from '.';

const meta = {
  title: 'Components/CustomerSay',
  component: CustomerSay,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],

  render: () => (
    <div className="w-[1280px]">
      <CustomerSay />
    </div>
  ),
} satisfies Meta<typeof CustomerSay>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
