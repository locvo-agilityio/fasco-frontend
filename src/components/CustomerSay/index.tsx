'use client';

import { ChevronLeft, ChevronRight } from 'lucide-react';

// Components
import { Button } from '@/components/common';
import CardReview from '../Card/CardReview';

// Images
import ImageReviewMiddle from '../../../public/assets/image-review-middle.webp';
import ImageReviewSecondary from '../../../public/assets/image-review-secondary.webp';

const CustomerSay = () => (
  <div className="hidden relative md:flex w-full pb-12 flex-col items-center justify-center bg-background-200 overflow-hidden">
    <h3 className="text-4xl font-volkhov text-center mt-24">
      This Is What Our Customers Say
    </h3>

    <p className="max-w-[614px] text-center text-gray-600 font-poppins mt-5">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Scelerisque duis
    </p>

    <div className="relative z-20 mt-16">
      <CardReview
        customClass="w-[840px] h-[400px] py-4 px-12 flex flex-row items-center gap-20 border-none"
        customImageClass="flex items-center w-[480px] h-[263px] rounded-regular"
        title="James K."
        rating={5}
        image={ImageReviewMiddle}
      />
    </div>

    <div className="absolute z-10 top-[52%] left-1/2 -translate-x-1/2 -translate-y-1/3 flex gap-56 opacity-80 scale-90">
      <CardReview
        customClass="w-[646px] h-[300px] py-10 px-12 flex flex-row items-center gap-4 border-none"
        customImageClass="flex items-center w-[377px] h-[193px] rounded-regular"
        title="James K."
        rating={5}
        image={ImageReviewSecondary}
      />

      <CardReview
        customClass="w-[646px] h-[300px] py-10 px-12 flex flex-row items-center gap-4 border-none"
        customImageClass="flex items-center w-[377px] h-[193px] rounded-regular"
        title="James K."
        rating={5}
        image={ImageReviewSecondary}
      />
    </div>

    <div className="flex gap-4 mt-12">
      <Button
        title="Prev Button"
        variant="ghost"
        color="primary"
        radius="full"
        className="w-12 h-12 min-w-12 border-none bg-white shadow-2xl disabled:cursor-not-allowed"
        disabled
      >
        <ChevronLeft className="text-primary-200" />
      </Button>
      <Button
        title="Next Button"
        variant="ghost"
        color="primary"
        radius="full"
        className="w-12 h-12 min-w-12 border-none bg-white shadow-2xl"
      >
        <ChevronRight className="text-primary-200" />
      </Button>
    </div>
  </div>
);

export default CustomerSay;
