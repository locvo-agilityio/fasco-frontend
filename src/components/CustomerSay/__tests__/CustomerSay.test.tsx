import { render } from '@testing-library/react';

// Components
import CustomerSay from '..';

describe('CustomerSay', () => {
  it('should render correctly', () => {
    const { container } = render(<CustomerSay />);

    expect(container).toMatchSnapshot();
  });
});
