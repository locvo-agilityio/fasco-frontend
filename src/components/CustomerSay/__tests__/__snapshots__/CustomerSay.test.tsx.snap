// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerSay should render correctly 1`] = `
<div>
  <div
    class="hidden relative md:flex w-full pb-12 flex-col items-center justify-center bg-background-200 overflow-hidden"
  >
    <h3
      class="text-4xl font-volkhov text-center mt-24"
    >
      This Is What Our Customers Say
    </h3>
    <p
      class="max-w-[614px] text-center text-gray-600 font-poppins mt-5"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Scelerisque duis
    </p>
    <div
      class="relative z-20 mt-16"
    >
      <div
        class="relative overflow-hidden text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-small rounded-large transition-transform-background motion-reduce:transition-none w-[840px] h-[400px] py-4 px-12 flex flex-row items-center gap-20 border-none"
        tabindex="-1"
      >
        <div
          class="relative p-3 flex-auto flex-col place-content-inherit align-items-inherit break-words text-left overflow-y-auto subpixel-antialiased flex items-center w-[480px] h-[263px] rounded-regular"
        >
          <img
            alt="James K."
            data-nimg="fill"
            decoding="async"
            sizes="50vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="p-3 flex overflow-hidden color-inherit subpixel-antialiased rounded-b-large w-full h-full flex-col items-start justify-center"
        >
          <p
            class="text-primary-200 font-poppins font-normal mb-7"
          >
            "You won't regret it. I would like to personally thank you for your outstanding product. Absolutely wonderful!"
          </p>
          <div
            class="w-full max-w-[230px] pb-7 border-b-1 border-b-primary-200"
          >
            <div
              class="flex gap-1"
            >
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
            </div>
          </div>
          <h4
            class="text-2xl font-volkhov text-center mt-4"
          >
            James K.
          </h4>
          <p
            class="mt-3 text-foreground-400 text-xs font-normal"
          >
            Traveler
          </p>
        </div>
      </div>
    </div>
    <div
      class="absolute z-10 top-[52%] left-1/2 -translate-x-1/2 -translate-y-1/3 flex gap-56 opacity-80 scale-90"
    >
      <div
        class="relative overflow-hidden text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-small rounded-large transition-transform-background motion-reduce:transition-none w-[646px] h-[300px] py-10 px-12 flex flex-row items-center gap-4 border-none"
        tabindex="-1"
      >
        <div
          class="relative p-3 flex-auto flex-col place-content-inherit align-items-inherit break-words text-left overflow-y-auto subpixel-antialiased flex items-center w-[377px] h-[193px] rounded-regular"
        >
          <img
            alt="James K."
            data-nimg="fill"
            decoding="async"
            sizes="50vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="p-3 flex overflow-hidden color-inherit subpixel-antialiased rounded-b-large w-full h-full flex-col items-start justify-center"
        >
          <p
            class="text-primary-200 font-poppins font-normal mb-7"
          >
            "You won't regret it. I would like to personally thank you for your outstanding product. Absolutely wonderful!"
          </p>
          <div
            class="w-full max-w-[230px] pb-7 border-b-1 border-b-primary-200"
          >
            <div
              class="flex gap-1"
            >
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
            </div>
          </div>
          <h4
            class="text-2xl font-volkhov text-center mt-4"
          >
            James K.
          </h4>
          <p
            class="mt-3 text-foreground-400 text-xs font-normal"
          >
            Traveler
          </p>
        </div>
      </div>
      <div
        class="relative overflow-hidden text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-small rounded-large transition-transform-background motion-reduce:transition-none w-[646px] h-[300px] py-10 px-12 flex flex-row items-center gap-4 border-none"
        tabindex="-1"
      >
        <div
          class="relative p-3 flex-auto flex-col place-content-inherit align-items-inherit break-words text-left overflow-y-auto subpixel-antialiased flex items-center w-[377px] h-[193px] rounded-regular"
        >
          <img
            alt="James K."
            data-nimg="fill"
            decoding="async"
            sizes="50vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="p-3 flex overflow-hidden color-inherit subpixel-antialiased rounded-b-large w-full h-full flex-col items-start justify-center"
        >
          <p
            class="text-primary-200 font-poppins font-normal mb-7"
          >
            "You won't regret it. I would like to personally thank you for your outstanding product. Absolutely wonderful!"
          </p>
          <div
            class="w-full max-w-[230px] pb-7 border-b-1 border-b-primary-200"
          >
            <div
              class="flex gap-1"
            >
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <svg
                class="lucide lucide-star fill-yellow-500 text-yellow-500"
                fill="none"
                height="19"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
            </div>
          </div>
          <h4
            class="text-2xl font-volkhov text-center mt-4"
          >
            James K.
          </h4>
          <p
            class="mt-3 text-foreground-400 text-xs font-normal"
          >
            Traveler
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex gap-4 mt-12"
    >
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:!bg-primary data-[hover=true]:!text-primary-foreground data-[hover=true]:!border-none text-secondary-400 border border-secondary-100 disabled:opacity-50 py-5 text-base rounded-full data-[pressed=true]:scale-[1] font-poppins w-12 h-12 min-w-12 border-none bg-white shadow-2xl disabled:cursor-not-allowed"
        disabled=""
        title="Prev Button"
        type="button"
      >
        <svg
          class="lucide lucide-chevron-left text-primary-200"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m15 18-6-6 6-6"
          />
        </svg>
      </button>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:!bg-primary data-[hover=true]:!text-primary-foreground data-[hover=true]:!border-none text-secondary-400 border border-secondary-100 disabled:opacity-50 py-5 text-base rounded-full data-[pressed=true]:scale-[1] font-poppins w-12 h-12 min-w-12 border-none bg-white shadow-2xl"
        title="Next Button"
        type="button"
      >
        <svg
          class="lucide lucide-chevron-right text-primary-200"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
